src/grep.rs

```diff
@@ -197,13 +203,16 @@ fn grep_offset(
 /// Run bgrep with the given options, outputting to the given `StdoutLock`.
 /// Error detail may be outputted to stderr.
 /// Returns whether there was a match.
-fn run_file(
-  stdout: &mut io::StdoutLock,
+fn run_file<O: Write, P: Into<PathBuf>, B: AsMut<Vec<u8>>>(
+  stdout: &mut O,
   options: &args::Options,
   pattern: &Regex,
-  path: PathBuf,
-  buffer: &mut Vec<u8>,
+  path: P,
+  mut buffer: B
```

Comment:
Shouldn't this be `buffer: &mut B`?

Reply:
You're correct. Sorry for my stupidity :(

Reply:
It was no stupidity, my friend. Be sure that I'm very glad with your contribution, these details are just to make sure everything is aligned.